packages/web-components/fast-components/src/select/select.styles.ts

```diff
@@ -0,0 +1,234 @@
+        border-radius: calc(var(--corner-radius) * 1px);
+        border: calc(var(--outline-width) * 1px) solid ${accentFillRestBehavior.var};
+        color: ${neutralForegroundRestBehavior.var};
+        contain: contents;
+        position: relative;
+        user-select: none;
+        min-width: 250px;
+    }
+
+    .listbox {
+        ${elevation}
+        background: ${neutralLayerFloatingBehavior.var};
+        border: calc(var(--outline-width) * 1px) solid ${neutralOutlineRestBehavior.var};
+        border-radius: calc(var(--corner-radius) * 1px);
+        box-sizing: border-box;
+        display: inline-flex;
+        flex-direction: column;
+        left: 0;
+        max-height: calc(var(--max-height) - (${heightNumber} * 1px));
+        padding: calc(var(--design-unit) * 1px) 0;
+        overflow-y: auto;
+        position: absolute;
+        width: 100%;
+        z-index: 1;
+    }
+
+    .listbox[hidden] {
+        display: none;
+    }
+
+    .control {
+        align-items: center;
+        box-sizing: border-box;
+        cursor: pointer;
+        display: flex;
+        font-size: var(--type-ramp-base-font-size);
+        font: inherit;
+        line-height: var(--type-ramp-base-line-height);
+        min-height: calc(${heightNumber} * 1px);
+        padding: calc(var(--design-unit) * 2.25px);
+        width: 100%;
+    }
+
+    :host(:not([disabled]):hover) {
+        background: ${neutralFillInputHoverBehavior.var};
+        border-color: ${accentFillHoverBehavior.var};
+    }
+
+    :host(:focus) .control {
+        border-color: ${accentFillFocusBehavior.var};
+    }
+
+    :host(:${focusVisible}) {
+        border-color: ${accentFillFocusBehavior.var};
+        box-shadow:
+            0 0 0 2px var(--background-color),
+            0 0 0 4px ${neutralFocusBehavior.var};
+    }
+
+    :host(:${focusVisible}) ::slotted(.selected[role="option"]:not([disabled])) {
+        box-shadow: 0 0 0 calc(var(--focus-outline-width) * 1px) inset ${
+            neutralFocusInnerAccentBehavior.var
+        };
+        border-color: ${neutralFocusBehavior.var};
+        background: ${accentFillHoverBehavior.var};
+        color: ${accentForegroundCutRestBehavior.var};
+    }
+
+    :host([disabled]) {
+        cursor: ${disabledCursor};
+        opacity: var(--disabled-opacity);
+    }
+
+    :host([disabled]) .control {
+        cursor: ${disabledCursor};
+    }
+
+    :host([disabled]:hover) {
+        background: ${neutralFillStealthRestBehavior.var};
+        color: ${neutralForegroundRestBehavior.var};
+        fill: currentcolor;
+    }
+
+    :host(:not([disabled])) .control:active {
+        background: ${neutralFillInputActiveBehavior.var};
+        border-color: ${accentFillActiveBehavior.var};
+    }
+
+    :host(.open.above) .listbox,
+    :host(.open.below) .control {
+        border-bottom-left-radius: 0;
+        border-bottom-right-radius: 0;
+    }
+
+    :host(.open.above) .control,
```

Comment:
same question here regarding attribute selectors.

Reply:
Updated to use attributes for `[open]`, and restructured to use `[position="above"]` and `[position="below"]`.